src/main/java/com/uniovi/util/IncidentSelector.java

```diff
@@ -27,30 +24,31 @@ public boolean isRelevant(Incident incident) {
 		return false;
 	}
 
+	@SuppressWarnings("unchecked")
 	private List<Function<Incident, Boolean>> relevanceConditions() {
 		
 		List<Function<Incident, Boolean>> conditions = new ArrayList<Function<Incident, Boolean>>();
 		
 		Reader reader = null;
-		JSONParser parser = new JSONParser();
 
 			try {
 				reader = new FileReader(confPath);
-				JSONObject jsonObject = (JSONObject) parser.parse(reader);
-				for (Object obj : jsonObject.keySet()) {
-					obtainCondition(conditions, jsonObject, (String) obj);
+				JsonNode incident = new ObjectMapper().readValue(reader, JsonNode.class);
+				/*for (Object obj : incident.fieldNames()) {
+					obtainCondition(conditions, incident, (String) obj);
+				}*/
+				while(incident.fieldNames().hasNext()){
```

Comment:
I'm guessing the problem is this loop. 
I think it could be that the iterator position is restarting every time, as you are getting the field names inside the condition. What I mean is that every time you do incident.fieldNames() the pointer is at -1, so hasNext() always returns true. I would try to do this:

> Iterator<String> itFieldNames = incident.fieldNames();
> while(itFieldNames.hasNext() ){
>            obtainCondition(conditions, incident, itFieldNames().next() );
> }